examples/src/main/java/io/dapr/examples/workflows/DemoWorkflowClient.java

```diff
@@ -24,29 +27,53 @@ public class DemoWorkflowClient {
 
   /**
    * The main method.
+   * 
    * @param args Input arguments (unused).
    * @throws InterruptedException If program has been interrupted.
    */
   public static void main(String[] args) throws InterruptedException {
     DaprWorkflowClient client = new DaprWorkflowClient();
 
     try (client) {
-      System.out.println("*****");
-      String instanceId = client.scheduleNewWorkflow(DemoWorkflow.class);
+      String separatorStr = "*******";
+      System.out.println(separatorStr);
+      String instanceId = client.scheduleNewWorkflow(DemoWorkflow.class, "input data");
       System.out.printf("Started new workflow instance with random ID: %s%n", instanceId);
 
-      System.out.println("Sleep and allow this workflow instance to timeout...");
-      TimeUnit.SECONDS.sleep(10);
+      System.out.println(separatorStr);
+      System.out.println("**GetInstanceMetadata:Running Workflow**");
+      WorkflowMetadata workflowMetadata = client.getInstanceMetadata(instanceId, true);
+      System.out.printf("Result: %s%n", workflowMetadata);
 
-      System.out.println("*****");
+      System.out.println(separatorStr);
+      System.out.println("**WaitForInstanceStart**");
+      try {
+        WorkflowMetadata waitForInstanceStartResult = 

```

Comment:
I have a different way implementing this. Would you like to have a look at that?
https://github.com/CSE-Neutrino/java-sdk/blob/cng/callActivity/examples/src/main/java/io/dapr/examples/workflows/DemoWorkflowClient.java#L45

Reply:
After implementing the `client.waitForInstanceCompletion` I don't think we need to wait using loop anymore